addons/AI_Assistant/src/presenter.js

```diff
@@ -0,0 +1,1556 @@
+                presenter.isFetching.request = false;
+                removeLoader();
+                presenter.sendRequest(response);
+            };
+        })
+        .catch(error => {
+            removeLoader();
+            presenter.handleInput('enable');
+            //createErrorMessage(error.status);
+            scrollToMessage(presenter.getLastMessage());
+            presenter.sendEventData('error', error.status);
+        });
+    };
+
+    //-===-===-===-===-====-===-===-===-===-\\
+
+    presenter.sendEventData = function (item, value) {
+        let eventData = presenter.createEventData(item, value);
+        if (presenter.playerController !== null) {
+            presenter.playerController.getEventBus().sendEvent('ValueChanged', eventData);
+        };
+    };
+
+    presenter.createEventData = function (item, value) {
+        let data = {
+            source: presenter.model.ID,
+            item: item,
+            value: value,
+        };
+        return data;
+    };
+
+    presenter.setVisibility = function (boolean) {
+        $(presenter.view).css('visibility', boolean ? 'visible' : 'hidden');
+        presenter.isVisible = boolean;
+        presenter.sendEventData('visibility', `${boolean}`)
+    };
+
+    presenter.show = function() {
+        if (presenter.classNames.show.length > 0) {
+            presenter.showAnimated();
+        } else {
+            presenter.setVisibility(true);
+        }
+    };
+
+    presenter.hide = function() {
+        if (presenter.classNames.hide.length > 0) {
+            presenter.hideAnimated();
+        } else {
+            presenter.setVisibility(false);
+        }
+    };
+
+    presenter.showAnimated = function() {
+        let closeName = presenter.classNames.hide.length > 0 ? presenter.classNames.hide : 'close';
+        let openName = presenter.classNames.show.length > 0 ? presenter.classNames.show : 'open';
+        presenter.view.classList.remove(closeName);
+        presenter.view.classList.add(openName);
+    }
+
+    presenter.hideAnimated = function() {
+        let closeName = presenter.classNames.hide.length > 0 ? presenter.classNames.hide : 'close';
+        let openName = presenter.classNames.show.length > 0 ? presenter.classNames.show : 'open';
+        presenter.view.classList.remove(openName);
+        presenter.view.classList.add(closeName);
+    }
+
+    presenter.setState = function (state) {
+        let parsedState = JSON.parse(state)
+
+        presenter.savedChatHistory = parsedState.savedChatHistory;
+        presenter.threadID = parsedState.threadID;
+        presenter.voiceMuted = parsedState.voiceMuted;
+    };
+
+    presenter.getState = function () {
+        presenter.pageChanged = true;
+        if (presenter.isFetching.request) presenter.removeMessage(presenter.getLastMessage());
+        presenter.stopReading();
+        return JSON.stringify({
+            // isVisible: presenter.isVisible,
```

Comment:
Dlaczego widoczność nie jest przekazywana w setState i getState? 
Jest to tutaj zakomentowane. Czyżby błąd?